__tests__/html/activityStatus.sendFailed.noEchoBack.html

```diff
@@ -0,0 +1,118 @@
+    <script>
+      function createDirectLineForTest() {
+        return testHelpers.createDirectLineWithTranscript([], {
+          overridePostActivity: activity => {
+            if (activity.type === 'message') {
+              return new Observable(observer => observer.next('a00001'));
+            } else {
+              return workingDirectLine.postActivity(activity);
+            }
+          }
+        });
+      }
+
+      function activityStatusInnerTextContained(expectedInnerTextPattern) {
+        return pageConditions.became(
+          `Activity status should contains "${expectedInnerTextPattern}"`,
+          () => pageElements.activityStatuses()[0].innerText.includes(expectedInnerTextPattern),
+          1000
+        );
+      }
+
+      function activityChannelDataStateBecame(expectedState) {
+        if (typeof expectedState === 'undefined') {
+          return pageConditions.became(
+            `Activity "channelData.state" should be undefined`,
+            () => !('state' in pageObjects.getActivities()[0].channelData),
+            1000
+          );
+        }
+
+        return pageConditions.became(
+          `Activity "channelData.state" should be "${expectedState}"`,
+          () => pageObjects.getActivities()[0].channelData?.state === expectedState,
+          1000
+        );
+      }
+
+      function activityChannelDataWebChatSendStatusBecame(expectedSendStatus) {
+        return pageConditions.became(
+          `Activity "channelData['webchat:send-status']" should be "${expectedSendStatus}"`,
+          () => pageObjects.getActivities()[0].channelData?.['webchat:send-status'] === expectedSendStatus,
+          1000
+        );
+      }
+
+      run(
+        async function () {
+          const clock = lolex.install();
+
+          WebChat.renderWebChat(
+            {
+              directLine: await createDirectLineForTest(),
+              store: testHelpers.createStore(),
+              styleOptions: { spinnerAnimationBackgroundImage: 'url(/assets/staticspinner.png)' }
+            },
+            document.getElementById('webchat')
+          );
+
+          await pageConditions.webChatRendered();
+
+          clock.tick(400);
+
+          // SETUP: Send a message
+          await pageConditions.uiConnected();
+          await pageObjects.sendMessageViaSendBox('Hello, World!', { waitForSend: false });
+
+          // THEN: `channelData.state` should be "sending".
+          await activityChannelDataStateBecame('sending');
+
+          // THEN: `channelData['webchat:send-status']` should be "sending".
+          await activityChannelDataWebChatSendStatusBecame('sending');
+
+          // THEN: The message should have status of "Sending".
+          await activityStatusInnerTextContained('Sending');
+
+          // WHEN: After 20 seconds.
+          clock.tick(20000);
+
+          // THEN: `channelData.state` should be "send failed".
+          await activityChannelDataStateBecame('send failed');
```

Comment:
This is the primary reason why we are no longer using the `channelData.state`. It expires after `styleOptions.sendTimeout` (normally 20s) and no way to turn it back.

This means, we cannot revive an activity after it become "send failed".

Thus, in previous version, we simply ignored "send failed" and assume it is "sending", or else, "sent". Thus, we are not recognizing the message failed to send due to reasons other than timeout.